.github/workflows/test.yml

```diff
@@ -1,31 +1,38 @@
-# This workflow uses actions that are not certified by GitHub.
-# They are provided by a third-party and are governed by
-# separate terms of service, privacy policy, and support
-# documentation.
-# This workflow will download a prebuilt Ruby version, install dependencies and run tests with Rake
-# For more information see: https://github.com/marketplace/actions/setup-ruby-jruby-and-truffleruby
-
 name: Testing
 
 on:
   push:
-    branches:
-      - master
   pull_request:
 
+# GitHub Actions notes
+# - outcome in step name so we can see it without having to expand logs
+# - every step must define a `uses` or `run` key
+
 jobs:
   test:
     name: Test with Ruby-${{ matrix.ruby }}
     runs-on: ubuntu-latest
+    timeout-minutes: 15
     strategy:
+      fail-fast: false
       matrix:
         # Due to https://github.com/actions/runner/issues/849, we have to use quotes for '3.0'
-        ruby: [2.6, 2.7, '3.0', 3.1, 'jruby-9.3']
+        ruby: [2.2, 2.3, 2.4, 2.5, 2.6, 2.7, '3.0', 3.1]
+        include:
+          - { ruby: jruby-9.3, allow-failure: true }
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - uses: ruby/setup-ruby@v1
+      continue-on-error: ${{ matrix.allow-failure || false }}
```

Comment:
This steps seems unlikely to error so I think it's enough just on `bundle exec rake`.

Reply:
My experience is that's it's common enough with JRuby/TruffleRuby when using the cache as this steps then runs `bundle install`

Reply:
I think that's more likely with `-head` than releases. But I could imagine a new Bundler release breaking something on existing jruby/truffleruby releases.
If every step is continue-on-error, isn't it better to put it on the whole job?